src/ParameterProvider/Aws/SecretsManager.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Dvsa\LaminasConfigCloudParameters\ParameterProvider\Aws;
+
+use Aws\Exception\AwsException;
+use Aws\SecretsManager\SecretsManagerClient;
+use Dvsa\LaminasConfigCloudParameters\Exception\ParameterProviderException;
+use Dvsa\LaminasConfigCloudParameters\ParameterProvider\ParameterProviderInterface;
+
+class SecretsManager implements ParameterProviderInterface
+{
+    protected SecretsManagerClient $secretsManagerClient;
+
+    public function __construct(
+        SecretsManagerClient $secretsManagerClient
+    ) {
+        $this->secretsManagerClient = $secretsManagerClient;
+    }
+
+    public function __invoke(string $id): array
+    {
+        try {
+            $result = $this->secretsManagerClient->getSecretValue([
+                'SecretId' => $id,
+            ]);
+        } catch (AwsException $e) {
+            throw new ParameterProviderException($e->getMessage(), $e->getCode(), $e);
+        }
+
+        if (isset($result['SecretString'])) {
+            $secret = $result['SecretString'];
+        } else {
+            $secret = base64_decode($result['SecretBinary']);
+        }
+
+        return json_decode($secret, true);
```

Comment:
There's a bit in the README where I've said we only support key/value pairs for secrets manager. 

Do you reckon it's worth supporting? Or could it be something for the future if someone puts a convincing use-case forward?

Reply:
What I have noticed though is the lack of JSON exception on invalid strings. I'll get the flag added.

While it would throw now, it'll be throwing a type exception as it's an `array` return type. So it'll improve the error message.

Reply:
Updated to throw exception on invalid JSON.